<?xml version="1.0" ?>
<launch>

  <arg name="number_of_cameras" default="9" />
  <arg name="hue_low" default="0"/>
  <arg name="hue_high" default="28"/>
  <arg name="sat_low" default="120"/>
  <arg name="sat_high" default="255"/>
  <arg name="val_low" default="50"/>
  <arg name="val_high" default="255"/>

  <arg name="p1_ip" default="192.168.1.121"/>
  <arg name="p2_ip" default="192.168.1.122"/>
  <arg name="p3_ip" default="192.168.1.123"/>
  <arg name="p4_ip" default="192.168.1.124"/>
  <arg name="p5_ip" default="192.168.1.125"/>
  <arg name="p6_ip" default="192.168.1.126"/>
  <arg name="p7_ip" default="192.168.1.127"/>
  <arg name="p8_ip" default="192.168.1.128"/>

  <!--arg name="hue_low" default="0"/>
  <arg name="hue_high" default="20"/>
  <arg name="sat_low" default="120"/>
  <arg name="sat_high" default="255"/>
  <arg name="val_low" default="50"/>
  <arg name="val_high" default="255"-->

  <arg name="detect_objects" default="False"/>
  <arg name="publish_images" default="True"/>
  <arg name="publish_segment_image" default="True"/>

  <include file="$(find move_to_see)/launch/picamera_frames.launch"/>

  <!-- <node machine="pi_camera_master" name="pi_camera_master_node" pkg="move_to_see" type="picam_array_master_node.py" output="screen" cwd="node">
    <param name="number_of_cameras" value="$(arg number_of_cameras)"/>
  </node>

  <machine name="pi_camera_master" address="192.168.1.19" user="pi" password="clusterhat"
    env-loader="~/picamera_ws/src/move_to_see_pi/move_to_see/launch/picamera_env.sh"
    timeout="30.0" /> -->

  <!-- <machine name="pi_camera_0" address="192.168.1.10" user="pi" password="clusterhat"
    env-loader="~/picamera_ws/src/move_to_see_pi/move_to_see/launch/picamera_env.sh"
    timeout="30.0" /> -->

  <!-- <node machine="pi_camera_0" name="pi_camera_node_0" pkg="move_to_see" type="picam_zero_capture_v2.py" output="screen" cwd="node">
    <param name="camera_index" value="0"/>
    <param name="camera_topic" value="/pi_camera_0"/>
    <param name="detect_objects" value="$(arg detect_objects)"/>
    <param name="publish_images" value="$(arg publish_images)"/>
    <param name="publish_segment_image" value="$(arg publish_segment_image)"/>

    <param name="hsv_hue_low" value="$(arg hue_low)"/>
    <param name="hsv_hue_high" value="$(arg hue_high)"/>
    <param name="hsv_val_low" value="$(arg val_low)"/>
    <param name="hsv_val_high" value="$(arg val_high)"/>
    <param name="hsv_sat_low" value="$(arg sat_low)"/>
    <param name="hsv_sat_high" value="$(arg sat_high)"/>

  </node> -->



  <machine name="pi_camera_1" address="$(arg p1_ip)" user="pi" password="clusterhat"
    env-loader="~/picamera_ws/src/move_to_see_pi/move_to_see/launch/picamera_env.sh"
    timeout="30.0" />

  <node machine="pi_camera_1" name="pi_camera_node_1" pkg="move_to_see" type="picam_zero_capture_v2.py" output="screen" cwd="node">
    <param name="camera_index" value="1"/>
    <param name="camera_topic" value="/pi_camera_1"/>
    <param name="detect_objects" value="$(arg detect_objects)"/>
    <param name="publish_images" value="$(arg publish_images)"/>
    <param name="publish_segment_image" value="$(arg publish_segment_image)"/>

    <param name="hsv_hue_low" value="$(arg hue_low)"/>
    <param name="hsv_hue_high" value="$(arg hue_high)"/>
    <param name="hsv_val_low" value="$(arg val_low)"/>
    <param name="hsv_val_high" value="$(arg val_high)"/>
    <param name="hsv_sat_low" value="$(arg sat_low)"/>
    <param name="hsv_sat_high" value="$(arg sat_high)"/>
  </node>

  <machine name="pi_camera_2" address="$(arg p2_ip)" user="pi" password="clusterhat"
    env-loader="~/picamera_ws/src/move_to_see_pi/move_to_see/launch/picamera_env.sh"
    timeout="30.0" />

  <node machine="pi_camera_2" name="pi_camera_node_2" pkg="move_to_see" type="picam_zero_capture_v2.py" output="screen" cwd="node">
    <param name="camera_index" value="2"/>
    <param name="camera_topic" value="/pi_camera_2"/>
    <param name="detect_objects" value="$(arg detect_objects)"/>
    <param name="publish_images" value="$(arg publish_images)"/>
    <param name="publish_segment_image" value="$(arg publish_segment_image)"/>

    <param name="hsv_hue_low" value="$(arg hue_low)"/>
    <param name="hsv_hue_high" value="$(arg hue_high)"/>
    <param name="hsv_val_low" value="$(arg val_low)"/>
    <param name="hsv_val_high" value="$(arg val_high)"/>
    <param name="hsv_sat_low" value="$(arg sat_low)"/>
    <param name="hsv_sat_high" value="$(arg sat_high)"/>
  </node>


  <machine name="pi_camera_3" address="$(arg p3_ip)" user="pi" password="clusterhat"
    env-loader="~/picamera_ws/src/move_to_see_pi/move_to_see/launch/picamera_env.sh"
    timeout="30.0" />

  <node machine="pi_camera_3" name="pi_camera_node_3" pkg="move_to_see" type="picam_zero_capture_v2.py" output="screen" cwd="node">
    <param name="camera_index" value="3"/>
    <param name="camera_topic" value="/pi_camera_3"/>
    <param name="detect_objects" value="$(arg detect_objects)"/>
    <param name="publish_images" value="$(arg publish_images)"/>
    <param name="publish_segment_image" value="$(arg publish_segment_image)"/>

    <param name="hsv_hue_low" value="$(arg hue_low)"/>
    <param name="hsv_hue_high" value="$(arg hue_high)"/>
    <param name="hsv_val_low" value="$(arg val_low)"/>
    <param name="hsv_val_high" value="$(arg val_high)"/>
    <param name="hsv_sat_low" value="$(arg sat_low)"/>
    <param name="hsv_sat_high" value="$(arg sat_high)"/>
  </node>


  <machine name="pi_camera_4" address="$(arg p4_ip)" user="pi" password="clusterhat"
    env-loader="~/picamera_ws/src/move_to_see_pi/move_to_see/launch/picamera_env.sh"
    timeout="30.0" />

  <node machine="pi_camera_4" name="pi_camera_node_4" pkg="move_to_see" type="picam_zero_capture_v2.py" output="screen" cwd="node">
    <param name="camera_index" value="4"/>
    <param name="camera_topic" value="/pi_camera_4"/>
    <param name="detect_objects" value="$(arg detect_objects)"/>
    <param name="publish_images" value="$(arg publish_images)"/>
    <param name="publish_segment_image" value="$(arg publish_segment_image)"/>

    <param name="hsv_hue_low" value="$(arg hue_low)"/>
    <param name="hsv_hue_high" value="$(arg hue_high)"/>
    <param name="hsv_val_low" value="$(arg val_low)"/>
    <param name="hsv_val_high" value="$(arg val_high)"/>
    <param name="hsv_sat_low" value="$(arg sat_low)"/>
    <param name="hsv_sat_high" value="$(arg sat_high)"/>
  </node>


  <machine name="pi_camera_5" address="$(arg p5_ip)" user="pi" password="clusterhat"
    env-loader="~/picamera_ws/src/move_to_see_pi/move_to_see/launch/picamera_env.sh"
    timeout="30.0" />

  <node machine="pi_camera_5" name="pi_camera_node_5" pkg="move_to_see" type="picam_zero_capture_v2.py" output="screen" cwd="node">
    <param name="camera_index" value="5"/>
    <param name="camera_topic" value="/pi_camera_5"/>
    <param name="detect_objects" value="$(arg detect_objects)"/>
    <param name="publish_images" value="$(arg publish_images)"/>
    <param name="publish_segment_image" value="$(arg publish_segment_image)"/>

    <param name="hsv_hue_low" value="$(arg hue_low)"/>
    <param name="hsv_hue_high" value="$(arg hue_high)"/>
    <param name="hsv_val_low" value="$(arg val_low)"/>
    <param name="hsv_val_high" value="$(arg val_high)"/>
    <param name="hsv_sat_low" value="$(arg sat_low)"/>
    <param name="hsv_sat_high" value="$(arg sat_high)"/>
  </node>


  <machine name="pi_camera_6" address="$(arg p6_ip)" user="pi" password="clusterhat"
    env-loader="~/picamera_ws/src/move_to_see_pi/move_to_see/launch/picamera_env.sh"
    timeout="30.0" />

  <node machine="pi_camera_6" name="pi_camera_node_6" pkg="move_to_see" type="picam_zero_capture_v2.py" output="screen" cwd="node">
    <param name="camera_index" value="6"/>
    <param name="camera_topic" value="/pi_camera_6"/>
    <param name="detect_objects" value="$(arg detect_objects)"/>
    <param name="publish_images" value="$(arg publish_images)"/>
    <param name="publish_segment_image" value="$(arg publish_segment_image)"/>

    <param name="hsv_hue_low" value="$(arg hue_low)"/>
    <param name="hsv_hue_high" value="$(arg hue_high)"/>
    <param name="hsv_val_low" value="$(arg val_low)"/>
    <param name="hsv_val_high" value="$(arg val_high)"/>
    <param name="hsv_sat_low" value="$(arg sat_low)"/>
    <param name="hsv_sat_high" value="$(arg sat_high)"/>
  </node>


  <machine name="pi_camera_7" address="$(arg p7_ip)" user="pi" password="clusterhat"
    env-loader="~/picamera_ws/src/move_to_see_pi/move_to_see/launch/picamera_env.sh"
    timeout="30.0" />

  <node machine="pi_camera_7" name="pi_camera_node_7" pkg="move_to_see" type="picam_zero_capture_v2.py" output="screen" cwd="node">
    <param name="camera_index" value="7"/>
    <param name="camera_topic" value="/pi_camera_7"/>
    <param name="detect_objects" value="$(arg detect_objects)"/>
    <param name="publish_images" value="$(arg publish_images)"/>
    <param name="publish_segment_image" value="$(arg publish_segment_image)"/>

    <param name="hsv_hue_low" value="$(arg hue_low)"/>
    <param name="hsv_hue_high" value="$(arg hue_high)"/>
    <param name="hsv_val_low" value="$(arg val_low)"/>
    <param name="hsv_val_high" value="$(arg val_high)"/>
    <param name="hsv_sat_low" value="$(arg sat_low)"/>
    <param name="hsv_sat_high" value="$(arg sat_high)"/>
  </node>


  <machine name="pi_camera_8" address="$(arg p8_ip)"  user="pi" password="clusterhat"
    env-loader="~/picamera_ws/src/move_to_see_pi/move_to_see/launch/picamera_env.sh"
    timeout="30.0" />

  <node machine="pi_camera_8" name="pi_camera_node_8" pkg="move_to_see" type="picam_zero_capture_v2.py" output="screen" cwd="node">
    <param name="camera_index" value="8"/>
    <param name="camera_topic" value="/pi_camera_8"/>
    <param name="detect_objects" value="$(arg detect_objects)"/>
    <param name="publish_images" value="$(arg publish_images)"/>
    <param name="publish_segment_image" value="$(arg publish_segment_image)"/>

    <param name="hsv_hue_low" value="$(arg hue_low)"/>
    <param name="hsv_hue_high" value="$(arg hue_high)"/>
    <param name="hsv_val_low" value="$(arg val_low)"/>
    <param name="hsv_val_high" value="$(arg val_high)"/>
    <param name="hsv_sat_low" value="$(arg sat_low)"/>
    <param name="hsv_sat_high" value="$(arg sat_high)"/>
  </node>


</launch>
